---
title: 'Reproduction of Critical Roles of the PPP and GLN3 in Isobutanol-specific Tolerance in Yeast'
  output: pdf_document:
    df_print: paged
---

We are aiming to reproduce the RNA-seq data processing of the paper 'Reproduction of Critical Roles of the Pentose Phosphate Pathways and GLN3 in Isotbutanol-specific Tolerance in Yeast'. For this, we got access to the raw data provided by the authors in ArrayExpress (link to the data here: https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-8175/samples/?keywords=&organism=Saccharomyces%20cerevisiae&exptype[0]=%22rna%20assay%22&exptype[1]=%22sequencing%20assay%22&array=&fbclid=IwAR3dXafs8wJy6KvOuBFWZ_VjdcdRLObAOHYBvS93zaXq3j9b1Xw60g4154M). 

We extracted the FASTq files following the instructions provided by Filip on Canvas. 

```{}
wget https://www.ebi.ac.uk/arrayexpress/files/E-MTAB-8175/E-MTAB-8175.sdrf.txt

less -S E-MTAB-8175.sdrf.txt

cat E-MTAB-8175.sdrf.txt | cut -f35 | tail -n +2 > links2download.txt #extracting the links to download the FASTq files
wget -N -i links2download.txt #download FASTq files 
 
```

As a result we obtained 16 FASTq files corresponding to our different samples. 

To understand better the context of the data, the samples are divided in wild-type (WT) strains and GLN3âˆ† (gln3) strains. For each genotype, we get two samples growing in normal conditions (add here information about "normal conditions" from the paper) and two samples growing in 1,3% v/v isobutanol. The RNA-seq experiment was duplicated for each of the samples, meaning that from 2 different cultures grown at same conditions we get 4 duplicates. 

We will also use the reference genome of S. cerevisiae to map our reads further on. To feth the reference genome, we went to the Sacharomyces Genome Database and download the same reference genome that they used for their experiment: S288C_reference_genome_R64-1-1-20110203

```{}
wget https://downloads.yeastgenome.org/sequence/S288C_reference/genome_releases/S288C_reference_genome_R64-1-1_20110203.tgz
tar -xvzf S288C_reference_genome_R64-1-1_20110203.tgz #to decompress the file

```

We will start by trying to align the reads to the reference genome that we downloaded. The alignment is made using TopHats. TopHats we will map the reads to the gene reference genome using the tool bowtie. However, to use the reference genome TopHats need an index for the reference genome. Therefore, we will first create this index by using bowtie2-build. 

```{}
bowtie2-build -f S288C_reference_sequence_R64-1-1_20110203.fsa bowtie_index/reference_genome
```
We allign the RNA-seq to the the S288C reference genome using tophat (version 2.0.13).  Using the following command: 

```{}
tophat -o alignment/dgln31 /reference_genome/bowtie_index/S288C raw_data/fasta_file_names/dgln31_1.fastq.gz raw_data/fasta_file_names/dgln31_2.fastq.gz
```




APPENDIX
List of programs and versions used and commands used to install it 


